test/integration/no-sampling.js

```diff
@@ -16,17 +17,23 @@ getPort().then(function (port) {
 
   test('should not sample', function (t) {
     var server = http.createServer(function (req, res) {
-      var buffers = []
-      var gunzip = zlib.createGunzip()
-      var unzipped = req.pipe(gunzip)
-
-      unzipped.on('data', buffers.push.bind(buffers))
-      unzipped.on('end', function () {
+      req = req.pipe(zlib.createGunzip()).pipe(ndjson.parse())
+
+      const received = {
+        metadata: 0,
+        transaction: 0
+      }
+      req.on('data', function (obj) {
+        const type = Object.keys(obj)[0]
+        received[type]++
+      })
+      req.on('end', function () {
+        t.equal(received.metadata, 1, 'expected 1 metadata to be sent')
+        t.equal(received.transaction, 20, 'expected 20 transactions to be sent')
         res.end()
         server.close()
-        var data = JSON.parse(Buffer.concat(buffers))
-        t.equal(data.transactions.length, 20, 'expect 20 transactions to be sent')
         t.end()
+        process.exit()
```

Comment:
Does this `process.exit()` need to be here? I'd rather not have these hiding in the tests as they could cause confusion when adding more tests to the file in the future, if you don't notice them.

Reply:
Fixed